packages/shorebird_cli/lib/src/shorebird_ios_release_version_mixin.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:propertylistserialization/propertylistserialization.dart';
+import 'package:shorebird_cli/src/command.dart';
+
+/// Helpers to determine the release and build number of an iOS app.
+mixin ShorebirdIosReleaseVersionMixin on ShorebirdCommand {
+  /// This key is a user-visible string for the version of the bundle. The
+  /// required format is three period-separated integers, such as 10.14.1. The
+  /// string can only contain numeric characters (0-9) and periods.
+  ///
+  /// See https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring
+  static const releaseVersionKey = 'CFBundleShortVersionString';
+
+  /// The version of the build that identifies an iteration of the bundle.
+  ///
+  /// See https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion
+  static const buildNumberKey = 'CFBundleVersion';
+
+  /// Checks the iOS Info.plist and referenced .xcconfig files to determine the
+  /// app's release version and build number.
+  Future<String> determineIosReleaseVersion() async {
+    final configPropertyRegex = RegExp(r'\$\((\w+)\)');
+
+    // TODO(bryanoltman): is it safe to assume "Runner" as the target name?
+    // See https://github.com/flutter/flutter/issues/9767
+    final plistPath = p.join(
+      Directory.current.path,
+      'ios',
+      'Runner',
+      'Info.plist',
+    );
+
+    final plist = PropertyListSerialization.propertyListWithString(
+      File(plistPath).readAsStringSync(),
+    ) as Map<String, Object>;
+    final pListVariables = _configVariables(
+      path: p.join(
+        Directory.current.path,
+        'ios',
+        'Flutter',
+        'Release.xcconfig',
+      ),
+    );
+    var releaseVersion = plist[releaseVersionKey] as String?;
+    var buildNumber = plist[buildNumberKey] as String?;
+
+    if (releaseVersion == null) {
+      throw Exception('Could not determine release version');
+    }
+
+    if (configPropertyRegex.hasMatch(releaseVersion)) {
+      releaseVersion = pListVariables[
```

Comment:
Kinda wild that we have to do our own plist variable substitution.  Feels like it belongs in a function somewhere. 🤷

Reply:
I think we'll eventually be able to rely on the xcode build to do this for us, but I think this should be robust enough for now.

Reply:
Yeah I think that makes sense

Reply:
Done.